Docs/sphinx_documentation/source/Faq.rst

```diff
@@ -0,0 +1,163 @@
+
+**A.** In the file ``amrex/Tools/GNUMake/Make.local`` you can specify your own compile
+commands by setting the variables ``CXX``, ``CC``, ``FC``, and ``F90``.
+An example can be found at `Specifying your own compiler`_ . Additional
+customizations are described in the file, ``amrex/Tools/GNUMake/Make.local.template``.
+In the same directory, ``amrex/Tools/GNUMake/README.md`` contains detailed
+information on compiler commands.
+
+.. _`Specifying your own compiler` : https://amrex-codes.github.io/amrex/docs_html/BuildingAMReX.html#specifying-your-own-compiler
+
+|
+
+**Q.** I'm having trouble compiling my code.
+
+**A.** AMReX developers have found that running the command ``make realclean`` can resolve
+many compilation issues.
+
+If you are working in an environment that uses
+a module system, please ensure you have the correct modules loaded. Typically, to do this,
+type ``module list`` at the command prompt.
+
+|
+
+**Q.** When I profile my code that uses GPUs with ``TINY_PROFILE=TRUE`` or ``PROFILE=TRUE``
+my timings are inconsistent.
+
+**A.** Due to the asynchronous nature of GPU execution, data collected by the built-in profilers
+may contain some variation. GPU profiling is an area of current development for the AMReX team.
+
+|
+
+**Q.** How do I know I am getting the right answer?
+
+**A.** AMReX provides support for verifying output with several tools. To briefly mention a few:
+
+- The :cpp:`print_state` command can be used to output the data of a single cell.
+- :cpp:`VisMF::Write` can be used to write MultiFab data to disk that can be viewed with `Amrvis`_.
+- :cpp:`amrex::Print()` is useful for printing
+  output when using multiple processes or threads as it prevents messages
+  from getting mixed up.
+- `fcompare`_ compares two plotfiles and reports absolute and relative error.
+
+Additional tools and discussion on this topic is contained
+in the section `Debugging`_.
+
+.. _`Debugging`: https://amrex-codes.github.io/amrex/docs_html/Basics.html#debugging
+
+.. _`Amrvis`: https://amrex-codes.github.io/amrex/docs_html/Visualization.html#sec-amrvis
+
+.. _`fcompare`: https://amrex-codes.github.io/amrex/docs_html/Post_Processing.html#fcompare
+
+|
+
+**Q.** What's the difference between :cpp:`copy` and :cpp:`parallelCopy`?
+
+**A.**
+
+|
+
+**Q.** How do I fill ghost cells?
+
+**A.**
+
+|
+
+**Q.** What's the difference between ``AmrCore`` and ``AmrLevel``? How do
+I decide which to use?
+
+**A.**
+
+|
+
+**Q.** For GPU usage, how can I perform explicit host to device and
+device to host copies without relying on managed memory?
+
+**A.** Use ``The_Pinned_Arena()`` and ``htod_memcpy()`` or ``dtoh_memcpy()``. See
+`Memory Allocation`_ in the AMReX Source Documentation.
+
+.. _`Memory Allocation`: https://amrex-codes.github.io/amrex/docs_html/GPU.html#memory-allocation
+
+|
+
+**Q.** How can I prevent a section of code from running on the GPU?
```

Comment:
I don't think this question is suitable for FAQ.

Reply:
Can we change the question to something that would be more suitable but still close? Or can we stress something in the answer, like "This should only be used for debugging purposes"  that would make it better?